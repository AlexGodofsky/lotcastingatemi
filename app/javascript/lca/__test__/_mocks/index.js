export const mockCharacter = {
  type: 'Character',
  attr_dexterity: 3, attr_wits: 3, attr_manipulation: 3, attr_stamina: 3, attr_strength: 2,
  abil_dodge: 4,  abil_integrity: 4, abil_socialize: 4,
  abil_melee: 2,
  abil_craft: [], abil_martial_arts: [], specialties: [],
  armor_weight: 'unarmored', armor_is_artifact: false,
  health_level_0s: 1, health_level_1s: 2, health_level_2s: 2, health_level_4s: 1, health_level_incap: 1,
  damage_bashing: 0, damage_lethal: 0, damage_aggravated: 0
}
