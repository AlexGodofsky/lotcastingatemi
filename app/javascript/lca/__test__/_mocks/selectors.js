// @flow
import { getPoolsAndRatings, getPenalties } from '../../selectors'
import type { fullChar } from '../../utils/propTypes/flow.js'

export const mockStateForCharacter = (character: fullChar) => ({ entities: {
  current: {
    characters: {
      [0]: {
        id: 0,
        ...character,
        merits: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
        weapons: [1, 2, 3],
        spells: [1, 2, 3],
      }
    },
    merits: {
      [1]:  { merit_name: 'ambidextrous' },
      [2]:  { merit_name: 'danger sense' },
      [3]:  { merit_name: 'fast reflexes' },
      [4]:  { merit_name: 'fleet of foot' },
      [5]:  { merit_name: 'mighty thew', rating: 2 },
      [6]:  { merit_name: 'pain tolerance' },
      [7]:  { merit_name: 'toxin resistance' },
      [8]:  { merit_name: 'chameleon' },
      [9]:  { merit_name: 'enhanced sense' },
      [10]: { merit_name: 'unusual hide', rating: 2 },
      [11]: { merit_name: 'living spirit cultivation' },
      [12]: { merit_name: 'vital focus cultivation' },
      [13]: { merit_name: 'hideous' },
    },
    spells: {
      [1]:  { name: 'invulnerable skin of bronze', control: true },
      [2]:  { name: 'impenetrable veil of night', control: true },
      [3]:  { name: 'rain of doom', control: true },
    },
    weapons: {
      [1]:  { name: 'light weapon',  weight: 'light',  attr: 'dexterity', damage_attr: 'strength', ability: 'melee', tags: ['melee', 'lethal'], character_id: 0,  },
      [2]:  { name: 'medium weapon', weight: 'medium', attr: 'dexterity', damage_attr: 'strength', ability: 'melee', tags: ['melee', 'lethal'], character_id: 0,  },
      [3]:  { name: 'heavy weapon',  weight: 'heavy',  attr: 'dexterity', damage_attr: 'strength', ability: 'melee', tags: ['melee', 'lethal'], character_id: 0,  },
    },
    charms: {
      [1]:  { ability: 'archery',       keywords: ['excellency'] },
      [2]:  { ability: 'athletics',     keywords: ['excellency'] },
      [3]:  { ability: 'awareness',     keywords: ['excellency'] },
      [4]:  { ability: 'brawl',         keywords: ['excellency'] },
      [5]:  { ability: 'bureaucracy',   keywords: ['excellency'] },
      [6]:  { ability: 'craft',         keywords: ['excellency'] },
      [7]:  { ability: 'dodge',         keywords: ['excellency'] },
      [8]:  { ability: 'integrity',     keywords: ['excellency'] },
      [9]:  { ability: 'investigation', keywords: ['excellency'] },
      [10]: { ability: 'larceny',       keywords: ['excellency'] },
      [11]: { ability: 'linguistics',   keywords: ['excellency'] },
      [12]: { ability: 'lore',          keywords: ['excellency'] },
      [13]: { ability: 'medicine',      keywords: ['excellency'] },
      [14]: { ability: 'melee',         keywords: ['excellency'] },
      [15]: { ability: 'occult',        keywords: ['excellency'] },
      [16]: { ability: 'performance',   keywords: ['excellency'] },
      [17]: { ability: 'presence',      keywords: ['excellency'] },
      [18]: { ability: 'resistance',    keywords: ['excellency'] },
      [19]: { ability: 'ride',          keywords: ['excellency'] },
      [20]: { ability: 'sail',          keywords: ['excellency'] },
      [21]: { ability: 'socialize',     keywords: ['excellency'] },
      [22]: { ability: 'stealth',       keywords: ['excellency'] },
      [23]: { ability: 'survival',      keywords: ['excellency'] },
      [24]: { ability: 'thrown',        keywords: ['excellency'] },
      [25]: { ability: 'war',           keywords: ['excellency'] },
      [26]: { charm_type: 'MartialArts' },
    }
  }
}})

export const mockGetPoolsAndRatings = (character: fullChar) =>
  getPoolsAndRatings(mockStateForCharacter(character), 0)

export const mockGetPenalties = (character: fullChar) =>
  getPenalties(mockStateForCharacter(character), 0)
