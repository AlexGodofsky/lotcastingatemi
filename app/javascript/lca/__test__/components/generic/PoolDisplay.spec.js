// @flow
declare var gen: Object
declare var check: Function

require('jasmine-check').install()

import React from 'react'
import renderer from 'react-test-renderer'

import PoolDisplay from '../../../components/generic/PoolDisplay.jsx'
import { SEED, genCharacter } from '../../_mocks'
import { mockGetPoolsAndRatings } from '../../_mocks/selectors.js'

describe('PoolDisplay', () => {
  check.it('renders all pools correctly', { times: 5, seed: SEED },
    gen.object(genCharacter), (mockCharacter) => {
      const pools = mockGetPoolsAndRatings(mockCharacter)
      const component = renderer.create(
        <div>
          { Object.keys(pools).map((p, i) => <PoolDisplay pool={ p } key={ i } />) }
        </div>
      )
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
})
