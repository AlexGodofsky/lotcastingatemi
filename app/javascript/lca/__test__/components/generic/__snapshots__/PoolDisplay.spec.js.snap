// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolDisplay renders all pools correctly for Solars 1`] = `
<div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
</div>
`;

exports[`PoolDisplay renders all pools correctly for Solars 2`] = `
<div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
</div>
`;

exports[`PoolDisplay renders all pools correctly for Solars 3`] = `
<div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
</div>
`;

exports[`PoolDisplay renders all pools correctly for Solars 4`] = `
<div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
</div>
`;

exports[`PoolDisplay renders all pools correctly for Solars 5`] = `
<div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
  <div
    className="PoolDisplay-root-1"
    onClick={[Function]}
  >
    <div
      className="PoolDisplay-label-2"
    >
      <span
        className="PoolDisplay-labelSpan-3"
      />
    </div>
    <div>
      <span
        className="PoolDisplay-pool-4"
      />
    </div>
  </div>
</div>
`;
